File: 1431.kids-with-the-greatest-number-of-candies.js
/*
 * @lc app=leetcode id=1431 lang=javascript
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {

    let ans = [];
    
    for(let i = 0; i < candies.length; i++){
        let count = 0;
        for(let j = 0; j < candies.length; j++){
            if(candies[i] + extraCandies >= candies[j] && i != j){
                count++
            }
        }

        if(count === candies.length-1){
            ans[i] = true;
        } else {
            ans[i] = false;
        }
    }

    return ans;
};
// @lc code=end

File: 1431.kids-with-the-greatest-number-of-candies.java
/*
 * @lc app=leetcode id=1431 lang=java
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        
        List<Boolean> ans = new ArrayList<Boolean>();

        int max = candies[0];

        for(int candy : candies){
            max = Math.max(max, candy);
        }

        for(int i = 0; i < candies.length; i++){
            if(candies[i] + extraCandies >= max){
                ans.add(true);
            } else {
                ans.add(false);
            }
        }

        return ans;
    }
}
// @lc code=end

